### PR TITLE
chore(github): github extension option

### PR DESCRIPTION
**Pull request by cathteng**
Option for the extensibility agent.

### CODE REVIEW

**Review by iamrajjoshi — APPROVED** [review]

**Review by mdtro — APPROVED** [review]

File: src/sentry/options/defaults.py [hunk 1]
```diff
@@ -552,6 +552,7 @@
     flags=FLAG_PRIORITIZE_DISK | FLAG_AUTOMATOR_MODIFIABLE,
 )
 register("github-login.organization", flags=FLAG_PRIORITIZE_DISK | FLAG_AUTOMATOR_MODIFIABLE)
+register("github-extension", default=False, flags=FLAG_MODIFIABLE_BOOL | FLAG_AUTOMATOR_MODIFIABLE)
```
Comment (cathteng) [inline src/sentry/options/defaults.py:555]:
```suggestion
register("github-extension.enabled", default=False, flags=FLAG_MODIFIABLE_BOOL | FLAG_AUTOMATOR_MODIFIABLE)
```

Comment (JoshFerge) [reply src/sentry/options/defaults.py:555]: did you not want it to specify enabled? also, did you not want this option configurable per org?

Comment (cathteng) [reply src/sentry/options/defaults.py:555]: this is not per org, github will not send us any information besides user information

Comment (JoshFerge) [reply src/sentry/options/defaults.py:555]: got it

Comment (cathteng) [reply src/sentry/options/defaults.py:555]: this is mainly for blocking the api from being hit rn in prod

**Pull request merged**